core/src/main/java/org/apache/calcite/runtime/JsonFunctions.java

```diff
@@ -631,6 +631,15 @@ public static String jsonRemove(JsonValueContext input, String... pathSpecs) {
     }
   }
 
+  public static Integer jsonStorageSize(Object input) {
```

Comment:
Please check how we handle JSON input on other JSON functions such as `JSON_LENGH`, `JSON_DEPTH`, the way we handle them changed after applying CALCITE-3017 fix.